src/idl_gen_general.cpp

```diff
@@ -828,6 +828,33 @@ void GenStruct(StructDef &struct_def, std::string *code_ptr) {
     code += ") + _bb.";
     code += lang_.get_bb_position;
     code += ", _bb)); }\n";
+
+    // recreate both methods for the prefixed size version
+    std::string ps_method_name = FunctionStart('G') + "etSizePrefixedRootAs" +
```

Comment:
This code has a lot of overlap with the code above, any ideas to refactor?

Reply:
I could see something like I've done in lines [1335+](https://github.com/google/flatbuffers/pull/4445/files/68be4420dda47e8d0600bb19691f03be71503a68#diff-d480fbc6a8bf3b7289fe45114ce2889bR1335). However the method bodies would depend on `i`, sort of forcing the for-switch anti-pattern.

Reply:
I've pushed a version with less overlap in e84452f.